sound/soc/soc-pcm.c

```diff
@@ -2282,42 +2282,91 @@ int dpcm_be_dai_trigger(struct snd_soc_pcm_runtime *fe, int stream,
 }
 EXPORT_SYMBOL_GPL(dpcm_be_dai_trigger);
 
+static int dpcm_fe_dai_trigger_fe_first(struct snd_pcm_substream *substream,
```

Comment:
@ranj063 phew, this makes my head burn! :) I mean "fe_trigger_fe_first" is a bit hard to understand. If we have a function to trigger "fe", one would expect this to trigger "fe" and not something else.

Anyways, the commit message does makes sense. I guess the big question are existing ok with change the the release ordering, or should DSP/machien drivers to have ability to specify PRE-versus-POST for each direction separately. I think you need to send to alsa-devel for this.

Reply:
@kv2019i unfortunately when a PCM is triggered it is always done through the FE. So the function to trigger FE takes care of triggering the corresponding BE's too. Now the ordering for PRE and POST work just fine for most cases. So I am a bit nervous about this change affecting other drivers. My first thought was to use the BESPOKE trigger but if you look at soc_pcm_bespoke_trigger(), it doesnt really have to do anything to do with the ordering of the FE/BE DAI's at all. In fact it just invokes the bespoke_triggers implemented by the dai drivers. 
And AFAICT, there are no drivers using this.

Anyway I will add some more context to the commit message and send this to alsa-devel